feat: Switch to opentelemetry tracer and harmonize service naming (#508)
kyma-project/istio: docs/release-notes/1.3.0.md

- Introduce conditions to Istio CR status [#468](https://github.com/kyma-project/istio/issues/468). Read [Istio Controller Overview](https://github.com/kyma-project/istio/blob/main/docs/user/00-10-overview-istio-controller.md) for more details

> - The Istio trace settings are now harmonized with the trace feature of the telemetry module and the semantic conventions of OpenTelemetry by having an improved `service.name` value attached to every span emitted by the Istio proxy.

Reviewer: ```suggestion
- To ensure consistency between the Istio trace settings, the Telemetry module's trace feature, and OpenTelemetry's semantic conventions, we have decided to attach the improved `service.name` value to every span emitted by the Istio proxy.
```

>   The service name determination now follows the same conventions as used in the telemetry module by first leveraging the `kubernetes.io/name` label. If not available, the `app` label is tried to be leveraged and then a fallback to the deployment name will happen.

Reviewer: ```suggestion
  To determine the service name, the Istio module now follows the same conventions as the Telemetry module. First, it tries to leverage the `kubernetes.io/name` label. If not available, it checks for the `app` label. If neither label is present, it falls back to using the Deployment's name.
```

>   Also, the namespace suffix is not attached anymore to the service name to be aligned and also follow the OpenTelemetry conventions.

Reviewer: ```suggestion
  Additionally, the namespace suffix is not attached anymore to the service name, which aligns with the conventions of OpenTelemetry.
```

>   Also, the Istio tracer is changed from the deprecated OpenCensus tracer to OpenTelemetry. As a side-effect the istio span name got improved by having the name prefixed with the operation name.
>   A name like `example.com" is now called `ingress example.com`. PR [#508](https://github.com/kyma-project/istio/pull/508). Read [Switch Istio Tracing to OTLP and align Istio service name](https://github.com/kyma-project/telemetry-manager/issues/557) for more details.

Reviewer: ```suggestion
   We have also updated the Istio tracer from the deprecated OpenCensus tracer to OpenTelemetry. This change has resulted in an improvement to the Istio span name, which is now prefixed with the operation name. For example, a name like `example.com" is changed to `ingress example.com`.
   To learn more, see the [PR #508](https://github.com/kyma-project/istio/pull/508) and read [Switch Istio Tracing to OTLP and align Istio service name](https://github.com/kyma-project/telemetry-manager/issues/557).
```
